src/dkg.rs

```diff
@@ -763,4 +789,39 @@ pub mod test {
         }
         assert_eq!(<C::Group as Group>::generator() * x, pub_key.to_element());
     }
+
+    pub fn test_reshare_threshold_limits<C: Ciphersuite>()
```

Comment:
Nice with the generic tests!

Nit: Pushing for the naming convention we haven't formalized yet

```suggestion
    pub fn reshare__should_fail_if_threshold_is_below_limit<C: Ciphersuite>()
```